src/extensions/default/HTMLCodeHints/main.js

```diff
@@ -746,7 +1223,14 @@ define(function (require, exports, module) {
         CodeHintManager.registerHintProvider(attrHints, ["html"], 0);
         NewFileContentManager.registerContentProvider(newDocContentProvider, ["html"], 0);
 
+        PreferencesManager.on("change", AllPreferences.EMMET, preferenceChanged);
+        preferenceChanged();
+
+        var emmetMarkupHints = new EmmetMarkupHints();
+        CodeHintManager.registerHintProvider(emmetMarkupHints, ["html", "php", "jsp"], 0);
```

Comment:
jsp is classses under HTML in phcode, so it doesnt need to be added here. see `src/language/languages.json` and search for `jsp`